### ISSUE TITLE
Unexpected noise in the echonet dataset (the mask should have taken care of it, but it doesn't)

### ISSUE THREAD

**Issue by Beep-Boop-Bopa** [echonet/dynamic#17]
This is what I did as follows:

1. Randomly select 100 videos out of Echonet.

2. Choose their very first frame (later on I repeated this process with frames other than the very first frame of the videos and saw the same results).

3. Apply a binary threshold to them s.t. any pixel with a value greater than 0 lights up. What we see here is a step like noise on the left and right edges of the ultrasound pie. **Should it be there?**

![image](https://user-images.githubusercontent.com/42900802/96503796-c8fa4e00-1221-11eb-8dc1-7299a01f9654.png)

4. Apply the mask given in the DICOM conversion notebook to them. The image on the left is the result from step 3, the image on the right is the same image after step 4. Please note the difference between the two.

![image](https://user-images.githubusercontent.com/42900802/96503441-42de0780-1221-11eb-8c77-d859e44af1ab.png)

All of the sampled images gave similar results. **If the mask provided had been used on the original echonet dataset, then we should not have seen any difference between the results of step 3 and 4** (the mask is applied after the frames have been resized to 112*112 size and no other modification is applied to the images after the application of the masks). What this means is that either noise was introduced later on to the videos or the dataset had a mask applied to it which was different from the one made available.

Could you provide any idea of why I am seeing these results?

The results could be reproduced with this code.

import glob
from random import sample 
import cv2
import torch,torchvision
import numpy as np

all_files = glob.glob('Videos/*'+'.avi')
Sampled_vids=sample(all_files,100)

def mask(output):
    dimension = output.shape[0]
    m1, m2 = np.meshgrid(np.arange(dimension), np.arange(dimension))
    mask = ((m1+m2)>int(dimension/2) + int(dimension/10)) 
    mask *=  ((m1-m2)<int(dimension/2) + int(dimension/10))
    mask = np.reshape(mask, (dimension, dimension)).astype(np.int8)
    maskedImage = cv2.bitwise_and(output, output, mask = mask)
    numpy_horizontal = np.hstack((output, maskedImage))
    return numpy_horizontal

for sample in Sampled_vids[:10]:
    Sample=cv2.VideoCapture(sample)
    ret, frame = Sample.read()
    frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
    (thresh, blackAndWhiteImage) = cv2.threshold(frame, 0, 255, cv2.THRESH_BINARY)
    cv2.imshow("result",mask(blackAndWhiteImage))
    cv2.waitKey(0)

cv2.destroyAllWindows()

![image](https://user-images.githubusercontent.com/42900802/96505453-21cae600-1224-11eb-923b-28ea8bb82a57.png)

**Comment by douyang** [comment]
I’m not quite sure what the question is. The released dataset is already masked and resized. The released dataset is the output of the DICOMtoAVI notebook. There is no need to mask again. The images attached seem to have undergone more thresholding which is not needed - can train directly on the images, the model will normalize prior to training loop.

**Comment by douyang** [comment]
Some of the images were masked with big_mask() instead of mask() then downsampled, which might create slightly different masks.
